lib/ClangImporter/ImportDecl.cpp

```diff
@@ -2486,14 +2486,114 @@ namespace {
         ctors.push_back(valueCtor);
       }
 
-      // Do not allow Swift to construct foreign reference types (at least, not
-      // yet).
       if (isa<StructDecl>(result)) {
+        // For c++ value types, import all the constructors
         for (auto ctor : ctors) {
           // Add ctors directly as they cannot always be looked up from the
           // clang decl (some are synthesized by Swift).
           result->addMember(ctor);
         }
+      } else if (isa<ClassDecl>(result)) {
+        // For C++ foreign reference types, generate a static factory method to
+        // construct the object and import it as a Swift initializer.
```

Comment:
Maybe this code snippet should be moved to a separate function for clarity.

Reply:
Also, let's add an experimental flag and move this codepath behind said flag.

Reply:
John had a recent PR adding such a flag: https://github.com/swiftlang/swift/pull/79728
You can look at that as an example.